src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/DirectoryControl.cs

```diff
@@ -389,7 +472,19 @@ public SearchOption SearchOption
 
         public override byte[] GetValue()
         {
-            _directoryControlValue = BerConverter.Encode("{i}", new object[] { (int)SearchOption });
+            int sizeEstimate = 8;
+            AsnWriter writer = new(AsnEncodingRules.BER, sizeEstimate);
```

Comment:
If you're trying to minimize overall allocation here, you can pool your writers.  Even if it's just

```C#
[ThreadStatic]
private static AsnWriter t_sharedWriter;
```

Whether you do one per type, or a central one, or whatever.  You could also do more complicated pooling, of course.